src/tests.c

```diff
@@ -4052,6 +4052,145 @@ void test_ecmult_multi(secp256k1_scratch *scratch, secp256k1_ecmult_multi_func e
     }
 }
 
+int test_ecmult_multi_random(secp256k1_scratch *scratch) {
+    secp256k1_scalar scalars[128];
+    secp256k1_scalar g_scalar, sc_tmp;
+    secp256k1_ge ge_tmp;
+    secp256k1_gej gejs[128];
+    secp256k1_gej result, computed;
+    secp256k1_ge ges[128];
+    ecmult_multi_data data;
+    int i;
+    /* Which multiplication function to use */
+    int fn = secp256k1_testrand_int(3);
+    secp256k1_ecmult_multi_func ecmult_multi = fn == 0 ? secp256k1_ecmult_multi_var :
+                                               fn == 1 ? secp256k1_ecmult_strauss_batch_single :
+                                               secp256k1_ecmult_pippenger_batch_single;
+    /* Simulate exponentially distributed num. */
+    int num_bits = 2 + secp256k1_testrand_int(6);
+    /* Number of (scalar, point) inputs (excluding g). */
+    int num = secp256k1_testrand_int((1 << num_bits) + 1);
+    /* Number of those which are nonzero. */
+    int num_nonzero = secp256k1_testrand_int(num + 1);
+    /* Whether we're aiming to create an input with nonzero result. */
+    int result_nonzero = secp256k1_testrand_bits(1);
+    /* Whether we will provide nonzero g multiplicand. */
+    int g_nonzero = num_nonzero == 0 ? result_nonzero :
+                    num_nonzero == 1 && !result_nonzero ? 1 :
+                    (int)secp256k1_testrand_bits(1);
+    /* Which g_scalar pointer to pass into ecmult_multi(). */
+    const secp256k1_scalar* g_scalar_ptr = (g_nonzero || secp256k1_testrand_bits(1)) ? &g_scalar : NULL;
+    /* How many scalars/gej pairs have been filled. */
+    int filled = 0;
+    /* How many EC multiplications were performed in this function. */
```

Comment:
"excluding ecmult_multi" if this is intentional. But it seems like it would be preferable to include `ecmult_multi` (the nonzero pairs) because `multi` appears to be a proxy for time and most of the multiplications will happen inside `ecmult_multi`. Probably worth to add a note on what `mults` is for exactly.

Reply:
Nice catch. That wasn't intentional (and also the reason why this test added so much runtime). I've fixed it.